countfiles/__main__.py

```diff
@@ -117,8 +122,10 @@ def main_flow(*args: [argparse_namespace_object, Union[bytes, str]]):
             show_2columns(sorted(data.items()))
         else:
             show_2columns(data.most_common())
+        # this part(table) is not tested at all, it returns None
```

Comment:
Do you think we should change `show_2columns()` into something like `create_table()`, make it just return a string (in order to allow for easier testing) and then print it here?

Reply:
return total_occurences will be sufficient for now.
this function has not changed since the creation.
I don't know how to test this.

Reply:
Well, probably we don't need to test it. But one way to do it would be to create a file containing the expected text for a given data, and then compare them.

Reply:
Yes, there is an opportunity to do as you suggest. I found it on the net.
```
from contextlib import redirect_stdout


def show_2columns(data):
    if not data:
        print("Oops! We have no data to show...\n")
        return

    max_word_width = 9  # default value, the minimum EXTENSION col. width
    total_occurences = 0
    for word, freq in data:
        total_occurences += freq
        max_word_width = max(len(word), max_word_width)

    total_occurences_width = len(str(total_occurences))
    if total_occurences_width < 5:
        total_occurences_width = 5

    header = f" {'EXTENSION'.ljust(max_word_width)} | {'FREQ.'.ljust(total_occurences_width)} "
    sep_left = (max_word_width + 2) * '-'
    sep_center = "+"
    sep_right = (total_occurences_width + 2) * '-'
    sep = sep_left + sep_center + sep_right
    print(header)
    print(sep)

    for word, freq in data:
        print(f" {word.ljust(max_word_width)} | {str(freq).rjust(total_occurences_width)} ")

    print(sep)
    line = f" {'TOTAL:'.ljust(max_word_width)} | {str(total_occurences).rjust(total_occurences_width)} "
    print(line)
    print(sep + "\n")

def get_file_extension(filepath: str) -> str:
    extension = os.path.splitext(filepath)[1][1:]
    if extension:
        return extension
    else:
        return '.'
    
from collections import Counter
import os
data = ['C:/Users/Net/Count-files/tests/data_for_tests\\html_file_for_tests.html',
             'C:/Users/Net/Count-files/tests/data_for_tests\\md_file_for_tests.md',
             'C:/Users/Net/Count-files/tests/data_for_tests\\no_extension',
             'C:/Users/Net/Count-files/tests/data_for_tests\\py_file_for_tests.py']
ext_for_counter = []
for some_path in data:
    extension = get_file_extension(some_path)
    if extension == '.':
        extension = '[no extension]'
    ext_for_counter.append(extension)
counter = Counter(ext_for_counter)

with open('test_2columns.txt', 'w') as f:
    with redirect_stdout(f):
        show_2columns(sorted(counter.items()))
```